rocketpy/simulation/multivariate_rejection_sampler.py

```diff
@@ -0,0 +1,259 @@
+"""
+Multivariate Rejection Sampling Module for RocketPy
+
+Notes
+-----
+This module is still under active development, and some features or attributes may
+change in future versions. Users are encouraged to check for updates and read the
+latest documentation.
+"""
+
+import json
+from dataclasses import dataclass
+from random import random
+
+from rocketpy._encoders import RocketPyEncoder
+
+
+@dataclass
+class SampleInformation:
+    """Sample information used in the MRS"""
+
+    inputs_json: dict = None
+    outputs_json: dict = None
+    probability_ratio: float = None
+    acceptance_probability: float = None
+
+
+class MultivariateRejectionSampler:
+    """Class that performs Multivariate Rejection Sampling (MRS) from MonteCarlo
+    results. The class currently assumes that all input variables are independent
+    when performing the
+
+    Attributes
+    ----------
+    """
+
+    def __init__(
+        self,
+        monte_carlo_filepath,
+        mrs_filepath,
+    ):
+        """Initializes Multivariate Rejection Sampler (MRS) class
+
+        Parameters
+        ----------
+        monte_carlo_filepath : str
+            Filepath prefixes to the files created from a MonteCarlo simulation
+            results and used as input for resampling.
+        mrs_filepath : str
+            Filepath prefix to MRS obtained samples. The files created follow the same
+            structure as those created by the MonteCarlo class but now containing the
+            selected sub-samples.
+
+        Returns
+        -------
+        None
+        """
+        self.monte_carlo_filepath = monte_carlo_filepath
+        self.mrs_filepath = mrs_filepath
+        self.distribution_dict = None
+        self.original_sample_size = 0
+        self.sup_ratio = 1
+        self.expected_sample_size = None
+        self.final_sample_size = None
+        self.input_variables_names = set()
+        self.output_variables_names = set()
+        self.all_sample_list = []
+        self.accepted_sample_list = []
+        self.__setup_input()
+        self.__load_output()
+
+    def __setup_input(self):
+        """Loads input information from monte carlo in a SampleInformation
+        object
+        """
+        input_filename = f"{self.monte_carlo_filepath}.inputs.txt"
```

Comment:
If possible, it is better practice to handle paths with the `pathlib` standard library. Check out the parallel monte carlo PR as an example of handling the `filename` in the MonteCarlo class.